pkg/cmd/server/server.go

```diff
@@ -90,8 +90,12 @@ type Config struct {
 	MetricsAPI   util.HTTPServerConfig
 
 	// Middleware for grpc
-	UnaryMiddleware     []grpc.UnaryServerInterceptor
-	StreamingMiddleware []grpc.StreamServerInterceptor
+	PrependUnaryMiddleware            []grpc.UnaryServerInterceptor
+	PrependStreamingMiddleware        []grpc.StreamServerInterceptor
+	ReplaceDefaultUnaryMiddleware     []grpc.UnaryServerInterceptor
```

Comment:
I'm still not sold on the naming here and we *definitely* need comments on the ordering that applies

Reply:
I've attempted to come up with another design in https://github.com/authzed/spicedb/pull/1063